source/opencascade/boundary_lib.cc

```diff
@@ -233,13 +233,13 @@ namespace OpenCASCADE
   ArclengthProjectionLineManifold<dim,spacedim>::pull_back(const Point<spacedim> &space_point) const
   {
     double t (0.0);
-#ifdef DEBUG
     ShapeAnalysis_Curve curve_analysis;
     gp_Pnt proj;
-    Assert(curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true)
-           < tolerance*length, ExcPointNotOnManifold(space_point));
+#ifdef DEBUG
+    Assert(curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true) <
+           tolerance*length, ExcPointNotOnManifold(space_point));
 #else
-    (void)space_point;
+    curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true);
```

Comment:
We could avoid this, at the price of having a warning at compilation time for release mode, where distance is not used...